From ba62dd32add2767d6d13f9abb82f43304418740f Mon Sep 17 00:00:00 2001
From: Andrew Borodin <aborodin@vmail.ru>
Date: Sun, 24 Jan 2021 14:19:22 +0300
Subject: [PATCH] Ticket #4181: FTBFS against NCurses on OS X 10.9.5.

lib/tty/tty-ncurses.h: define _XOPEN_SOURCE_EXTENDED to use cchar_t,
getcchar(), setcchar().

Signed-off-by: Andrew Borodin <aborodin@vmail.ru>
[Retrieved from:
https://github.com/MidnightCommander/mc/commit/ba62dd32add2767d6d13f9abb82f43304418740f]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 lib/tty/tty-ncurses.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/tty/tty-ncurses.h b/lib/tty/tty-ncurses.h
index 9962c7c413..d75df9533a 100644
--- a/lib/tty/tty-ncurses.h
+++ b/lib/tty/tty-ncurses.h
@@ -2,6 +2,11 @@
 #ifndef MC__TTY_NCURSES_H
 #define MC__TTY_NCURSES_H
 
+/* for cchar_t, getcchar(), setcchar() */
+#ifndef _XOPEN_SOURCE_EXTENDED
+#define _XOPEN_SOURCE_EXTENDED
+#endif
+
 #ifdef USE_NCURSES
 #ifdef HAVE_NCURSES_CURSES_H
 #include <ncurses/curses.h>
